inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp

```diff
@@ -0,0 +1,234 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <thread>
+#include <future>
+
+#include "shared_test_classes/subgraph/basic_lstm.hpp"
+#include "behavior/ov_infer_request/io_tensor.hpp"
+#include "functional_test_utils/ov_tensor_utils.hpp"
+
+namespace ov {
+namespace test {
+
+void OVInferRequestIOTensorTest::SetUp() {
+    InferRequestTests::SetUp();
+    req = execNet.create_infer_request();
+    input = execNet.get_parameters().front();
+    input_name = input->get_friendly_name();
+    output = execNet.get_results().front();
+    output_name = output->input_value(0).get_node()->get_friendly_name();
+}
+
+void OVInferRequestIOTensorTest::TearDown() {
+    req = {};
+    input = {};
+    output = {};
+    InferRequestTests::TearDown();
+}
+
+TEST_P(OVInferRequestIOTensorTest, Cancreate_infer_request) {
+    auto req = execNet.create_infer_request();
+}
+
+TEST_P(OVInferRequestIOTensorTest, failToSetNullptrForInput) {
+    ASSERT_THROW(req.set_tensor(input_name, {}), ov::Exception);
+}
+
+TEST_P(OVInferRequestIOTensorTest, failToSetNullptrForOutput) {
+    auto req = execNet.create_infer_request();
+    ASSERT_THROW(req.set_tensor(output_name, {}), ov::Exception);
+}
+
+TEST_P(OVInferRequestIOTensorTest, setNotAllocatedInput) {
+    auto tensor = create_and_fill_tensor(input->get_element_type(), input->get_shape());
+    req.set_tensor(input_name, tensor);
```

Comment:
looks like it's allocated

Reply:
Seems that we should throw an exception here. 
Or we should define some additional behavior here.